Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -0,0 +1,288 @@
+		}
+		else
+		{
+			TSharedPtr<FRepLayout> RepLayout = NetDriver->GetStructRepLayout(Struct);
+
+			RepLayout_SerializePropertiesForStruct(*RepLayout, ValueDataWriter, PackageMap, const_cast<uint8*>(Data), bHasUnmapped);
+		}
+
+		Schema_AddPayload(Object, Id, ValueDataWriter);
+	}
+	else if (UBoolProperty* BoolProperty = Cast<UBoolProperty>(Property))
+	{
+		Schema_AddBool(Object, Id, (std::uint8_t)BoolProperty->GetPropertyValue(Data));
+	}
+	else if (UFloatProperty* FloatProperty = Cast<UFloatProperty>(Property))
+	{
+		Schema_AddFloat(Object, Id, FloatProperty->GetPropertyValue(Data));
+	}
+	else if (UDoubleProperty* DoubleProperty = Cast<UDoubleProperty>(Property))
+	{
+		Schema_AddDouble(Object, Id, DoubleProperty->GetPropertyValue(Data));
+	}
+	else if (UInt8Property* Int8Property = Cast<UInt8Property>(Property))
+	{
+		Schema_AddInt32(Object, Id, (std::int32_t)Int8Property->GetPropertyValue(Data));
+	}
+	else if (UInt16Property* Int16Property = Cast<UInt16Property>(Property))
+	{
+		Schema_AddInt32(Object, Id, (std::int32_t)Int16Property->GetPropertyValue(Data));
+	}
+	else if (UIntProperty* IntProperty = Cast<UIntProperty>(Property))
+	{
+		Schema_AddInt32(Object, Id, IntProperty->GetPropertyValue(Data));
+	}
+	else if (UInt64Property* Int64Property = Cast<UInt64Property>(Property))
+	{
+		Schema_AddInt64(Object, Id, Int64Property->GetPropertyValue(Data));
+	}
+	else if (UByteProperty* ByteProperty = Cast<UByteProperty>(Property))
+	{
+		Schema_AddUint32(Object, Id, (std::uint32_t)ByteProperty->GetPropertyValue(Data));
+	}
+	else if (UUInt16Property* UInt16Property = Cast<UUInt16Property>(Property))
+	{
+		Schema_AddUint32(Object, Id, (std::uint32_t)UInt16Property->GetPropertyValue(Data));
+	}
+	else if (UUInt32Property* UInt32Property = Cast<UUInt32Property>(Property))
+	{
+		Schema_AddUint32(Object, Id, UInt32Property->GetPropertyValue(Data));
+	}
+	else if (UUInt64Property* UInt64Property = Cast<UUInt64Property>(Property))
+	{
+		Schema_AddUint64(Object, Id, UInt64Property->GetPropertyValue(Data));
+	}
+	else if (UObjectPropertyBase* ObjectProperty = Cast<UObjectPropertyBase>(Property))
+	{
+		UnrealObjectRef ObjectRef = SpatialConstants::NULL_OBJECT_REF;
+
+		UObject* ObjectValue = ObjectProperty->GetObjectPropertyValue(Data);
+		if (ObjectValue != nullptr)
+		{
+			FNetworkGUID NetGUID = PackageMap->GetNetGUIDFromObject(ObjectValue);
+			if (!NetGUID.IsValid())
+			{
+				if (ObjectValue->IsFullNameStableForNetworking())
+				{
+					NetGUID = PackageMap->ResolveStablyNamedObject(ObjectValue);
+				}
+			}
+			ObjectRef = UnrealObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(NetGUID));
+			if (ObjectRef == SpatialConstants::UNRESOLVED_OBJECT_REF)
+			{
+				// A legal static object reference should never be unresolved.
+				check(!ObjectValue->IsFullNameStableForNetworking());
+				UnresolvedObjects.Add(ObjectValue);
+				ObjectRef = SpatialConstants::NULL_OBJECT_REF;
+			}
+		}
+
+		Schema_AddObjectRef(Object, Id, ObjectRef);
```

Comment:
In general, it's probably a better idea to not make C API helper functions look exactly the same as built in C API functions. I think it's a bit friendlier to the reader to make it obvious what code is owned by the C API and what code is owned by the GDK.